### PR TITLE
EZP-27363: Register PDF files via REST API

### PR DESCRIPTION
**Pull request by DominikaK**
https://jira.ez.no/browse/EZP-27363

### CODE REVIEW

**Comment by DominikaK** [comment]
@adamwojs, this is based on a document you provided, could do with a look from you.

File: docs/cookbook/creating_content_with_binary_attachments_via_rest_api.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
+# Creating content with binary attachments via REST API
+
+## Description
+
+This recipe illustrates how to create content via the REST API on the example of uploading an image file.
+
+## Solution
+
+The creation of the draft is done with a POST request on /api/ezp/v2/content/objects
+with a body containing the required and structured data.
```
Comment (adamwojs) [inline docs/cookbook/creating_content_with_binary_attachments_via_rest_api.md]: We should link to the documentation here: https://doc.ezplatform.com/en/1.11/guide/field_type_reference/#creating-content-data-property Reader will be able to find more details there.

File: docs/cookbook/creating_content_with_binary_attachments_via_rest_api.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+As for the response, it's possible to use either JSON or XML in input.
+The following examples use JSON.
```
Comment (adamwojs) [inline docs/cookbook/creating_content_with_binary_attachments_via_rest_api.md]: This snippet assumes that HTTP Basic Auth is enabled.  https://doc.ezplatform.com/en/1.11/api/general_rest_usage/#http-basic-authentication

File: docs/cookbook/creating_content_with_binary_attachments_via_rest_api.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+$curl = curl_init();
+
+curl_setopt_array($curl, array(
+   CURLOPT_URL => "$base_url/api/ezp/v2/content/objects/$id/versions/1",
+   CURLOPT_RETURNTRANSFER => true,
+   CURLOPT_ENCODING => "",
+   CURLOPT_MAXREDIRS => 10,
+   CURLOPT_TIMEOUT => 30,
+   CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
+   CURLOPT_CUSTOMREQUEST => "POST",
```
[the remark this answers is not in the document]
Comment (andrerom) [reply docs/cookbook/creating_content_with_binary_attachments_via_rest_api.md]: solved in 8a8497a

File: docs/cookbook/creating_content_with_binary_attachments_via_rest_api.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+curl_setopt_array($curl, array(
+    CURLOPT_URL => "$base_url/api/ezp/v2/content/objects",
+    CURLOPT_RETURNTRANSFER => true,
+    CURLOPT_ENCODING => "",
+    CURLOPT_MAXREDIRS => 10,
+    CURLOPT_TIMEOUT => 30,
+    CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
+    CURLOPT_CUSTOMREQUEST => "POST",
+    CURLOPT_POSTFIELDS => json_encode($data),
```
[the remark this answers is not in the document]
Comment (andrerom) [reply docs/cookbook/creating_content_with_binary_attachments_via_rest_api.md]: solved in 8a8497a

**Pull request merged**